source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -0,0 +1,168 @@
+#include "extensions/filters/listener/http_inspector/http_inspector.h"
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/network/listen_socket.h"
+#include "envoy/stats/scope.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/common/assert.h"
+#include "common/http/headers.h"
+
+#include "extensions/transport_sockets/well_known_names.h"
+
+#include "absl/strings/match.h"
+#include "nghttp2/nghttp2.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace HttpInspector {
+
+Config::Config(Stats::Scope& scope)
+    : stats_{ALL_HTTP_INSPECTOR_STATS(POOL_COUNTER_PREFIX(scope, "http_inspector."))} {}
+
+thread_local uint8_t Filter::buf_[Config::MAX_INSPECT_SIZE];
```

Comment:
does this have to be `thread_local`?

Reply:
No. IMO, I think making `buf_` as a data member is ok. I followed the pattern in tls inspector which makes the `buf_` `thread_local`.

Reply:
Hmm, not sure why buf_ was `static thread_local` in TLS inspector too. From #3062 looks like due to some perf optimization, do you remember the reason? @PiotrSikora @alyssawilk @mattklein123

Reply:
I think it was made `static thread_local` (with the emphasis on `static`) to avoid allocating this buffer over and over again.

Reply:
Ah that makes sense. @crazyxy can you add a comment for the reason?

Reply:
Sure